changelog/changed-factorize-MemoryInterface-slight-variations.md

```diff
@@ -0,0 +1 @@
+Factorize the variants of `MemoryInterface` where only the return type changes into `MemoryInterface` with an associated type.
```

Comment:
I think this is not true any more, we no longer have the associated type :)

Reply:
That’s right !